### PR TITLE
Fix incorrect HA-flow protected path allocation

### LINKED ISSUE #5165: HA-flow protected path is equal to main path
**Steps to reproduce:**
1. Create ha flow with protected path via `POST /v2/ha-flows` with enabled protected path
2. check that main ha-flow path and protected path does not intersect 
You can do it by `GET /v2/ha-flows/{ha_flow_id}/paths` when issue https://github.com/telstra/open-kilda/issues/5148 will be ready or you can check it directly in orient DB

**Expected result:**
Paths do not intersect (excluding first and last switch)

**Actual result:**
paths are equal

### PR DESCRIPTION
**Pull request by niksv**
Closes #5165

### CODE REVIEW

**Review by Fedichk — APPROVED** [review]

File: src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/FlowPathBuilder.java [hunk 1]
```diff
@@ -89,22 +89,31 @@ public boolean isSamePath(Path path, FlowPath flowPath) {
...
+     * Check whether the ha-path and ha-flow path overlap.
+     *
+     * @param haPath the ha-path to evaluate.
+     * @param haFlowPath the ha-flow path to evaluate.
+     */
+    public boolean arePathsOverlapped(HaPath haPath, HaFlowPath haFlowPath) {
+        for (Path subPath : haPath.getSubPaths()) {
+            Set<Segment> pathSegments = buildSegmentSet(subPath);
+            for (FlowPath haFlowSubPath : haFlowPath.getSubPaths()) {
+                Set<Segment> haPathSegments = buildSegmentSet(haFlowSubPath);
```
[the remark this answers is not in the document]
Comment (pablomuri) [reply src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/FlowPathBuilder.java:108]: agree

File: src-java/base-topology/base-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/FlowPathBuilderTest.java [hunk 2]
```diff
@@ -16,40 +16,51 @@
...
 public class FlowPathBuilderTest {
+    public static final SwitchId SWITCH_ID_1 = new SwitchId(1);
+    public static final SwitchId SWITCH_ID_2 = new SwitchId(2);
+    public static final SwitchId SWITCH_ID_3 = new SwitchId(3);
+    public static final Switch SWITCH_1 = Switch.builder().switchId(SWITCH_ID_1).build();
+    public static final Switch SWITCH_2 = Switch.builder().switchId(SWITCH_ID_2).build();
+    public static final Switch SWITCH_3 = Switch.builder().switchId(SWITCH_ID_3).build();
+    public static final PathId PATH_ID_1 = new PathId("test_path_id");
+    public static final String HA_SUB_FLOW = "ha_sub_flow";
+
```
Comment (pablomuri) [inline src-java/base-topology/base-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/FlowPathBuilderTest.java:63]: why are those public?

File: src-java/base-topology/base-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/FlowPathBuilderTest.java [hunk 3]
```diff
@@ -68,7 +79,7 @@ public void setUp() {
     }
 
     @Test
-    public void shouldDetectSameSwitchPaths() {
+    public void detectSameSwitchPathsTest() {
```
Comment (pablomuri) [inline src-java/base-topology/base-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/FlowPathBuilderTest.java:82]: adding "Test" at the end is redundant. Same for the rest

Comment (niksv) [reply src-java/base-topology/base-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/FlowPathBuilderTest.java]: fixed

**Review by pablomuri — APPROVED** [review]

**Pull request merged**